Source/WebKit/WebProcess/WebPage/WebPage.cpp

```diff
@@ -1779,12 +1780,55 @@ void WebPage::platformDidReceiveLoadParameters(const LoadParameters& loadParamet
 }
 #endif
 
+void WebPage::loadRequestByCreatingNewLocalFrameOrConvertingRemoteFrame(LocalFrameCreationParameters&& localFrameCreationParameters, LoadParameters&& loadParameters)
+{
+    RefPtr frame = WebProcess::singleton().webFrame(localFrameCreationParameters.frameIdentifier);
+    if (frame && frame->coreFrame()) {
+        loadRequest(WTFMove(loadParameters));
+        return;
+    }
+
+    // If there is an existing RemoteFrame, let's remove it and later convert it to a LocalFrame
+    RefPtr<WebFrame> parentWebFrame;
+    RefPtr<WebCore::RemoteFrame> coreRemoteFrame;
+    std::optional<ScopeExit<Function<void()>>> invalidator;
+    if (frame) {
+        coreRemoteFrame = frame->coreRemoteFrame();
+        auto* parent = coreRemoteFrame->tree().parent();
+        if (!parent) {
+            ASSERT_NOT_REACHED();
+            return;
+        }
+
+        parentWebFrame = WebProcess::singleton().webFrame(parent->frameID());
+        if (!parentWebFrame) {
+            ASSERT_NOT_REACHED();
+            return;
+        }
+
+        parent->tree().removeChild(*coreRemoteFrame);
+        coreRemoteFrame->disconnectOwnerElement();
+        if (auto* remoteFrameClient = static_cast<WebRemoteFrameClient*>(&coreRemoteFrame->client()))
```

Comment:
We should add the right template specializations so we can use downcast.

Reply:
I don't think so in this case.  This follows the pattern of static_cast<WebFrameLoaderClient

Reply:
I don’t understand your point. Why don’t we want a safer cast here? Or why can’t we?
The fact that existing code is unsafe doesn’t sound like a good reason to not make code safer.